File: leetCode/1768. Merge Strings Alternately/solution.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let newWord = "";
    let index = 0;
    while ( index < word1.length || index < word2.length ){
        newWord += index < word1.length ? word1[index] : "";
        newWord += index < word2.length ? word2[index] : "";
        index++;
    }
    return newWord;
};

File: leetCode/1768. Merge Strings Alternately/solution.py

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        newWordArray = []
        index = 0
        moreWord1 = True
        moreWord2 = True
        while moreWord1 or moreWord2:
            try:
                newWordArray.append(word1[index])
            except:
                moreWord1 = False
            try:
                newWordArray.append(word2[index])
            except:
                moreWord2 = False
            index += 1
        return "".join(newWordArray)